python/requirements.in

```diff
@@ -4,5 +4,6 @@ cachetools
 email-validator
 fastapi
 mangum
+pyjwt
```

Comment:
There are many JWT implementations in Python:
- https://github.com/jpadilla/pyjwt
- https://github.com/mpdavis/python-jose
- https://github.com/lepture/authlib

What led you to select PyJWT?

Reply:
Real reason: PyJWT was used extensively in code examples I found online, and appears to be a health project (4.1k stars on GitHub). Previously I have used python-jose so let's compare:

PyJWT makes fetching the signing key for a given JWT very easy. 

```
jwks_client = jwt.PyJWKClient(self.jwks_uri)
signing_key = jwks_client.get_signing_key_from_jwt(jwt)
```

python-jose does not provide this method, so you need to do something like this:

```
async with httpx.AsyncClient() as client:
    resp = await client.get(jwks_uri)
jwks_data = resp.json()
header = jwt.get_unverified_header(token.access_token)
signing_key = next(data for data in jwks_data['keys'] if data['kid'] == header['kid'])
```

Note an obvious benefit of the latter method is that as you're free to use whatever library you want to fetch the `jwks_data`, you can do this asynchronously, which is not available in PyJWT. Other than this one thing it's pretty much as wash between PyJWT and python-jose, as the method for validating a JWT with the signing key is identical between the two. 

authlib seemed heavyweight for the task at hand. python-jose is already a dependency of `architecture-functions` module, so if we convert to python-jose by using the code above (async or not), that would do the job with less dependencies, very doing the job with less lines of code.

Reply:
48baf5f changes to using python-jose instead of PyJWT. I don't really have much skin in the game here, either works, and I'm sure authlib does too.

Reply:
A search would suggest the CVE score is authlib has two (CVE-2022-39174, CVE-2022-39175 https://docs.authlib.org/en/latest/community/security.html), python-jose one ([CVE-2016-7036](https://www.opencve.io/cve/CVE-2016-7036)) and PyJWT 2 ([CVE-2022-29217](https://www.opencve.io/cve/CVE-2022-29217), [CVE-2017-11424](https://www.opencve.io/cve/CVE-2017-11424)).  

Looking on snyk, limiting on issues reported on the pip installable packags, pyjwt and authlib have more reported vulnerabilities than python-jose.
https://security.snyk.io/vuln/pip?search=python-jose   (1 issue)
https://security.snyk.io/vuln/pip?search=pyjwt  (3 issues)
https://security.snyk.io/vuln/pip?search=authlib  (6 issues)

Reply:
Thanks. These are all sensible choices; it's nice that python-nose is also the one with the least vulnerabilities, historically.